### PR TITLE
[MIG] membership_withdrawal: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
Depends on:

- [x] membership_extension

Features functionality present in https://github.com/OCA/crm/pull/160

cc @Tecnativa

### CODE REVIEW

**Review by luismontalba — APPROVED** [review]
Tested on runbot

**Comment by pedrobaeza** [comment]
Have you taken into account https://github.com/OCA/vertical-association/pull/22 remarks?

**Comment by chienandalu** [comment]
@pedrobaeza I took the ones answered for good. I've made a little fix though.

File: membership_withdrawal/README.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+* Withdrawal reason: Why member requested withdrawal
+* Withdrawal date: When member requested withdrawal
+* Date cancel: Date when membership period must finish
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/208/8.0
```
Comment (pedrobaeza) [inline membership_withdrawal/README.rst]: Incorrect target

File: membership_withdrawal/README.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/{project_repo}/issues>`_. In case of trouble, please
```
Comment (pedrobaeza) [inline membership_withdrawal/README.rst]: Put project

File: membership_withdrawal/README.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+check there if your issue has already been reported. If you spotted it first,
+help us smash it by providing detailed and welcomed feedback.
```
Comment (pedrobaeza) [inline membership_withdrawal/README.rst]: Don't change this text. It was correct.

File: membership_withdrawal/models/membership_line.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+from odoo import fields, models
+
+
+class MembershipLine(models.Model):
+    _inherit = "membership.membership_line"
+
+    date_withdrawal = fields.Date(
+        string="Withdrawal date",
+        help="Date when member requested membership withdrawl")
```
Comment (pedrobaeza) [inline membership_withdrawal/models/membership_line.py]: s/withdrawl/withdrawal

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by chienandalu** [comment]
@pedrobaeza Changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please squash your last commits for merging

**Comment by chienandalu** [comment]
@pedrobaeza All green now

**Pull request merged**